src/cluster_legacy.c

```diff
@@ -4776,7 +4776,7 @@ void clusterFailoverReplaceYourPrimary(void) {
 
     /* 3) Update state and save config. */
     clusterUpdateState();
-    clusterSaveConfigOrDie(1);
+    clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG | CLUSTER_TODO_FSYNC_CONFIG);
```

Comment:
There is a clusterDoBeforeSleep right below this, typically they seem all grouped together.

Reply:
updated!